ribbon/src/main/java/feign/ribbon/RibbonClient.java

```diff
@@ -26,6 +28,16 @@
  */
 public class RibbonClient implements Client {
 
+  private static final LoadingCache<String, LBClient> LBCLIENT_CACHE = CacheBuilder.newBuilder()
+      .build(new CacheLoader<String, LBClient>() {
+        @Override
+        public LBClient load(String clientName) throws Exception {
```

Comment:
No problem with the changes suggested but a couple of counterpoints

*  Using a single RibbonClient across the entire VM might be a bit limiting as this forces the use of only a single Client delegate. Which I believe is probably fine if you are using the Default.Client. But maybe not possible with a custom Client delegate.
* Not using a static map will mean that the error will still occur if someone disregards the advice or needs to use different delegates - This would be my main concern.
* If we create a synchronized method getOrCreateLBClient this means that all request via the RibbonClient will incur the penalty ( we can avoid this with some double locking and volatile)